src/spacy/app.clj

```diff
@@ -52,8 +58,53 @@
          (assoc :uris {::submit-session
                        (bidi/path-for routes ::submit-session :event-id event-id)})))))))
 
-(defn submit-session [req]
-  (resp/redirect (bidi/path-for routes ::index)))
+(defn- random-uuid []
+  (java.util.UUID/randomUUID))
+
+;; TODO - probably implemented in Crux DB
+(defn update-state [state {:keys [title description] :as session}]
+  (let [id (random-uuid)
+        current-user "joy" ;; TODO - retrieve from header
+        new-session {:sponsor current-user :session (assoc session :id id)}
+        new-facts [[id :session-title title]
+                   [id :description description]
+                   [current-user :has-suggested id]]]
+    (-> state
+        (update-in [:waiting-queue] #(concat % [new-session]))
+        (update-in [:facts] #(concat % new-facts)))))
+
```

Comment:
> TODO - probably implemented in Crux DB

I like it as it is, that is as a data transformation with neither side effects nor library dependencies. I think we can keep it in its current shape and move it to, say, `spacy.domain`.

Then we could introduce an adapter that gets the results of the pure operation and persists them in Crux, away from the domain namespace.

We can sketch it later this week.